src/inputs/pipe.c

```diff
@@ -302,6 +304,51 @@ parse_progress(struct input_metadata *m, char *progress)
   m->song_length = (end - start) * 10 / 441; // Convert to ms based on 44100
 }
 
+static void
+parse_volume(char *volume)
+{
+  char *value_begin;
+  char *value_next;
+  char *value_end;
+  float airplay_volume;
+
+  if (!(value_begin = strtok_r(volume, ",", &value_next)))
+    return;
+
+  airplay_volume = strtof(value_begin, &value_end);
+
+  if ((errno == ERANGE) || (value_begin == value_end))
+    {
+      char no_number[] = "First token is not a number.";
```

Comment:
You'll notice that forked-daapd doesn't have mid-function declarations, so please remove